### PR TITLE
Move alarms converter code to a separate file

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
Signed-off-by: Pradeep Kumar K J <pradeepkumar.kj@softwareag.com>

## Proposed changes

Refactor the `c8y alarms` code and move the `alarms` code to a separate file.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_mapper/src/c8y/converter.rs [hunk 1]
```diff
@@ -153,7 +152,7 @@ where
         .try_into()
         .expect("topics that mapper should subscribe to");
 
-        topic_filter.add_all(CumulocityMapper::subscriptions(&operations).unwrap());
+        let () = topic_filter.add_all(CumulocityMapper::subscriptions(&operations).unwrap());
```
Comment (matthiasbeyer) [inline crates/core/tedge_mapper/src/c8y/converter.rs:155]: Please remove.

File: crates/core/tedge_mapper/src/c8y/converter.rs [hunk 2]
```diff
@@ -348,7 +347,7 @@ where
     async fn try_convert(&mut self, message: &Message) -> Result<Vec<Message>, ConversionError> {
         match &message.topic {
             topic if topic.name.starts_with("tedge/measurements") => {
-                self.size_threshold.validate(message)?;
+                let () = self.size_threshold.validate(message)?;
```
Comment (matthiasbeyer) [inline crates/core/tedge_mapper/src/c8y/converter.rs:350]: Please remove.

File: crates/core/tedge_mapper/src/c8y/converter.rs [hunk 3]
```diff
@@ -933,10 +778,10 @@ mod tests {
         let operation_logs = OperationLogs::try_new(log_dir.path().to_path_buf()).unwrap();
 
         let now = std::time::Instant::now();
-        super::execute_operation("5", "sleep", "sleep_one", &operation_logs)
+        let () = super::execute_operation("5", "sleep", "sleep_one", &operation_logs)
```
Comment (matthiasbeyer) [inline crates/core/tedge_mapper/src/c8y/converter.rs:781]: Please remove.

File: crates/core/tedge_mapper/src/c8y/converter.rs [hunk 4]
```diff
@@ -933,10 +778,10 @@ mod tests {
...
             .await
             .unwrap();
-        super::execute_operation("5", "sleep", "sleep_two", &operation_logs)
+        let () = super::execute_operation("5", "sleep", "sleep_two", &operation_logs)
```
Comment (matthiasbeyer) [inline crates/core/tedge_mapper/src/c8y/converter.rs:784]: Please remove.

**Review by matthiasbeyer — DISMISSED** [review]
Don't forget to rebase once approved, the last commit does not have the signoff.

File: crates/core/tedge_mapper/src/c8y/converter.rs (hunk 4 above)
Comment (PradeepKiruvale) [reply crates/core/tedge_mapper/src/c8y/converter.rs]: removed

File: crates/core/tedge_mapper/src/c8y/converter.rs (hunk 3 above)
Comment (PradeepKiruvale) [reply crates/core/tedge_mapper/src/c8y/converter.rs:781]: removed

File: crates/core/tedge_mapper/src/c8y/converter.rs (hunk 2 above)
Comment (PradeepKiruvale) [reply crates/core/tedge_mapper/src/c8y/converter.rs]: removed

File: crates/core/tedge_mapper/src/c8y/converter.rs (hunk 1 above)
Comment (PradeepKiruvale) [reply crates/core/tedge_mapper/src/c8y/converter.rs]: removed

**Review by albinsuresh — APPROVED** [review]

**Pull request merged**